sys/sys/efiio.h

```diff
@@ -35,17 +35,17 @@
 struct efi_get_table_ioc
 {
 	struct uuid uuid;	/* UUID to look up */
-	void *ptr;		/* Pointer to table in KVA space */
+	void * __kerncap ptr;	/* Pointer to table in KVA space */
```

Comment:
I wonder if this should instead be `kvaddr_t`

Reply:
That sounds sensible, except for being an ABI break for freebsd32. I had proposed vm_offset_t on a review related to this ioctl, though nothing really came of that. `void *` is definitely wrong and you need to cast it to an integer to have any hope of using it correctly.

Reply:
I don't see any consumers in tree. I've asked @kevans91 about it on IRC

Reply:
IMO it's obviously right and it would be easy to add a compat IOCTL for 32-bit if anyone cared.

Reply:
I think you misunderstand. `kvaddr_t` is _always_ a uint64_t, so this breaks the 32-bit syscall interface regardless of whether it's native or compat. And in fact you currently _do_ need to have a compat shim due to using a pointer, but using `kvaddr_t` would _remove_ the need for that. I.e. it's a "you need to rebuild userspace issue", not a "there now needs to be a compat shim" issue.

Reply:
kib@ thinks it should be vm_paddr_t which wouldn't break the ABI (except on i386 where this is unused). The comment is also wrong in that this is a physical address.

Reply:
It also breaks n32 MIPS. Not that anyone likely cares.

Reply:
The consensus on IRC is that we should delete EFIIOC_GET_TABLE as it has no users and an extremely dubious interface. Hopefully we'll have an upstream commit we could cherry-pick tomorrow.

Reply:
I'm glad attempt #2 at removing it has succeeded :)

Reply:
It looks like you can cherry pick 9be9771c875259b9c99414feb3833c5af03baa8b from upstream in place of the changes to this file.